tests/src/test/java/org/eclipse/hono/tests/mqtt/MqttPublishTestBase.java

```diff
@@ -374,6 +375,51 @@ public void testUploadMessagesWithWrongContentTypeSendsErrors(final VertxTestCon
         testUploadMessageWithInvalidContentType(ctx, tenantId, deviceId, deviceId);
     }
 
+    /**
+     * Verifies that a message from a device with a payload exceeding the maximum size cannot be consumed and causes
+     * the connection to be closed.
+     *
+     * @param ctx The vert.x test context.
+     * @throws InterruptedException if test execution gets interrupted.
+     */
+    @Test
+    public void testMessageWithPayloadExceedingMaxSize(final VertxTestContext ctx) throws InterruptedException {
+
+        final String tenantId = helper.getRandomTenantId();
+        final String deviceId = helper.getRandomDeviceId(tenantId);
+        final Tenant tenant = new Tenant();
+        final VertxTestContext setup = new VertxTestContext();
+
+        helper.registry.addDeviceForTenant(tenantId, tenant, deviceId, "secret")
+                .onComplete(setup.succeedingThenComplete());
+
+        assertThat(setup.awaitCompletion(IntegrationTestSupport.getTestSetupTimeout(), TimeUnit.SECONDS)).isTrue();
+        if (setup.failed()) {
+            ctx.failNow(setup.causeOfFailure());
+            return;
+        }
+
+        final byte[] payloadBytes = IntegrationTestSupport.getPayload(ServiceConfigProperties.DEFAULT_MAX_PAYLOAD_SIZE);
+        final Promise<Void> clientClosedPromise = Promise.promise();
+        // GIVEN a connected device
+        connectToAdapter(IntegrationTestSupport.getUsername(deviceId, tenantId), "secret")
+                .compose(connAck -> {
+                    mqttClient.closeHandler(v -> clientClosedPromise.complete());
+                    // WHEN a device publishes a message with a payload exceeding the max size
+                    return send(tenantId, deviceId, Buffer.buffer(payloadBytes), false, null);
```

Comment:
shouldn't a payload of size ServiceConfigProperties.DEFAULT_MAX_PAYLOAD_SIZE actually be accepted?

Reply:
We are setting the `MqttServerOptions#maxMessageSize` on the MQTT server here. This size consists of the payload size plus the size of the variable header, containing the topic name (e.g. `command/DEFAULT_TENANT/4711/res/1010f8ab0b53-bd96-4d99-9d9c-56b868474a6a/200/?on-error=disconnect`). 
So, only payloads that are a number of bytes less in size than the `DEFAULT_MAX_PAYLOAD_SIZE` are currently accepted. 

How about we account for these extra bytes needed for the topic name by adding 150 bytes to the max message size?
`mqttServerOptions.setMaxMessageSize(getConfig().getMaxPayloadSize() + 150);`

Reply:
sounds good to me